internal/storage/bucket_handle_test.go

```diff
@@ -256,3 +257,90 @@ func (t *BucketHandleTest) TestCreateObjectMethodWhenGivenGenerationObjectNotExi
 	AssertEq(nil, obj)
 	AssertTrue(strings.Contains(err.Error(), "Error 412: Precondition failed"))
 }
+
+func (t *BucketHandleTest) TestGetProjectionValueEqualZero() {
+	proj := getProjectionValue(0)
+
+	AssertEq(storage.Projection(1), proj)
+}
+
+func (t *BucketHandleTest) TestGetProjectionValueEqualOne() {
+	proj := getProjectionValue(1)
+
+	AssertEq(storage.Projection(2), proj)
+}
+
+func (t *BucketHandleTest) TestGetProjectionValueForDefault() {
+	proj := getProjectionValue(7)
+
+	AssertEq(storage.Projection(1), proj)
+}
+
+func (t *BucketHandleTest) TestListObjectMethodWithPrefixObjectExist() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "gcsfuse/",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "ContinuationToken",
+			MaxResults:               7,
+			ProjectionVal:            0,
+		})
+
+	AssertEq(nil, err)
+	AssertEq(TestObjectName, obj.Objects[0].Name)
+	AssertEq(TestObjectGeneration, obj.Objects[0].Generation)
+	AssertEq("gcsfuse/", obj.CollapsedRuns[0])
+}
+
+func (t *BucketHandleTest) TestListObjectMethodWithPrefixObjectDoesNotExist() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "PrefixObjectDoesNotExist",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "ContinuationToken",
+			MaxResults:               7,
+			ProjectionVal:            0,
+		})
+
+	AssertEq(nil, err)
+	AssertEq(nil, obj.Objects)
+	AssertEq(nil, obj.CollapsedRuns)
+}
+
+func (t *BucketHandleTest) TestListObjectMethodWithIncludeTrailingDelimiter() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "gcsfuse/",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: true,
```

Comment:
add testcase for includetrailingdelimiter false scenario. I see include trailingdelimiter is handled: https://github.com/GoogleCloudPlatform/gcsfuse/blob/9b96e9dc586959661a1ad54622bbb7419e7e3e68/vendor/github.com/fsouza/fake-gcs-server/fakestorage/object.go#L316

Reply:
fixed